docs/source/json-api/index.rst

```diff
@@ -862,41 +862,40 @@ output a series of JSON documents, each ``payload`` formatted according
 to :doc:`lf-value-specification`::
 
     {
-        "events": [
-            {
-                "created": {
+        "events": [{
+            "created": {
+                "observers": [],
+                "agreementText": "",
+                "payload": {
                     "observers": [],
-                    "agreementText": "",
-                    "payload": {
-                        "observers": [],
-                        "issuer": "Alice",
-                        "amount": "999.99",
-                        "currency": "USD",
-                        "owner": "Alice"
-                    },
-                    "signatories": [
-                        "Alice"
-                    ],
-                    "contractId": "#1:0",
-                    "templateId": "b70bbfbc77a4790f66d4840cb19f657dd20848f5e2f64e39ad404a6cbd98cf75:Iou:Iou"
+                    "issuer": "Alice",
+                    "amount": "999.99",
+                    "currency": "USD",
+                    "owner": "Alice"
                 },
-                "matchedQueries": [
-                    1,
-                    2
-                ]
-            }
-        ]
+                "signatories": ["Alice"],
+                "contractId": "#1:0",
+                "templateId": "eb3b150383a979d6765b8570a17dd24ae8d8b63418ee5fd20df20ad2a1c13976:Iou:Iou"
+            },
+            "matchedQueries": [1, 2]
+        }]
     }
 
 where ``matchedQueries`` indicates the 0-based indices into the request
 list of queries that matched this contract.
 
-When the stream reaches the end of contracts that existed when the
-request started, you'll receive a special message indicating the start
-of "live" updates.  For example, you might use it to turn off an initial
-"loading" indicator::
+Every ``events`` block following the end of contracts that existed when
+the request started includes an ``offset``.  The stream is guaranteed to
+send an offset immediately at the beginning of this "live" data, which
+may or may not contain any ``events``; if it does not contain events and
+no events were emitted before, it may be ``null`` instead of a string.
+For example, you might use it to turn off an initial "loading"
+indicator::
 
-    {"live": true}
+    {
+        "events": [],
+        "offset": "2"
```

Comment:
Can we say something about the type of the offset? Is it always a string? Or shall we treat it as something opaque whose only purpose it to pass it to the next call of this endpoint?

Reply:
The previous paragraph mentions the rules for what it might be; do you think there is too much detail there?

Reply:
Oh right
> it may be `null` instead of a string.

Can we guarantee that it's either a string or null? Or could it be something else? Particularly on other ledgers which can't represent the offset by a single number.

Reply:
Even for ledgers that want to use more structure, the ledger API requires them to string-encode it, and we merely pass-through for the purpose of #4509.

https://github.com/digital-asset/daml/blob/4d734939c9490bd0cdfd5faf36351d1fcba12e73/ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/ledger_offset.proto#L14-L22

Reply:
Then it makes sense. Thansl.